algorithms/jobtitle_sampler/sampler.py

```diff
@@ -0,0 +1,21 @@
+import random
+
+def get_num_lines():
+    with open(filename) as f:
+        num_lines = sum(1 for line in f)
+    return num_lines
+
+def reservoir_sample(sample_count, filename):
```

Comment:
Some comments:
1. It would make sense to make the random seed injectable, right?
2. I would like to see a test for this. If the random seed is injectable, it would make sense to call this twice on the same dataset with different (handpicked) seeds and assert you get different lists each time
3. This, as written, isn't really a job title sampler. It looks like a generic file-based reservoir sampler; it only acts as a job title reservoir sampler when called with a filename that contains job titles. This by itself is a good thing, but I do think this is misnamed and will lead to confusion among readers of the code.